client/data/newsletter-categories/use-newsletter-categories-query.tsx

```diff
@@ -0,0 +1,58 @@
+import { useQuery, UseQueryResult } from '@tanstack/react-query';
+import request from 'wpcom-proxy-request';
+
+type NewsletterCategoryQueryProps = {
+	blogId: number;
+};
+
+type NewsletterCategoryQueryResponse = {
+	newsletter_categories: NewsletterCategoryResponse[];
+};
+
+type NewsletterCategories = {
+	newsletterCategories: NewsletterCategory[];
+};
+
+type NewsletterCategoryResponse = {
+	term_id: number;
+	name: string;
+	description: string;
+	order: number;
+};
+
+type NewsletterCategory = {
+	termId: number;
+	name: string;
+	description: string;
+	order: number;
+};
+
+const convertNewsletterCategoryResponse = ( response: NewsletterCategoryQueryResponse ) => {
+	return {
+		newsletterCategories: response.newsletter_categories.map(
+			( { term_id, name, description, order }: NewsletterCategoryResponse ) => {
+				return {
+					termId: term_id,
+					name,
+					description,
+					order,
+				};
+			}
+		),
+	};
+};
+
+const useNewsletterCategories = ( {
+	blogId,
+}: NewsletterCategoryQueryProps ): UseQueryResult< NewsletterCategories > => {
```

Comment:
And if you apply the change from the comment above you don't need a new type to define the return shape, just `NewsletterCategory[]` would work.

Reply:
well, the returned value has this shape:
```
export type NewsletterCategories = {
	newsletterCategories: NewsletterCategory[];
};
```
So I guess I still need this type here.

Reply:
I wonder if we need the `newsletterCategories` in the output shape of the query. It seems to be a bit much to do:

```ts
const { data } = useNewsletterCategories();
const newsLetterCategories = data?.newsLetterCategories;
```

Without this extra level, we could do:

```ts
const { data: newsLetterCategories } = useNewsletterCategories();
```

Instead of converting `newsletter_categories` to camelcase, I'd just return the array.

Reply:
It's similar to a previous discussion on [this comment](https://github.com/Automattic/wp-calypso/pull/80101#discussion_r1281211041). Juan's point was more into expanding for pagination or any additional properties in the future, like the way we do on `useSubscribersQuery`. On the other hand, I'm not sure if that will be the case for this endpoint. There won't be many categories to require pagination, and it doesn't seem to be the idea for the UI, so maybe we are overthinking it.